rust/core-lib/src/tabs.rs

```diff
@@ -349,16 +349,23 @@ impl Documents {
         self.plugins.toggle_tracing(enabled);
     }
 
-    fn save_trace<P: AsRef<Path>>(&self, destination: P, frontend_samples: &Value) {
+    fn save_trace<P: AsRef<Path>>(&self, destination: P, frontend_samples: Value) {
         use xi_trace_dump::*;
 
         let mut frontend_trace = chrome_trace::decode(frontend_samples)
             .unwrap_or(Vec::with_capacity(0));
 
-        let mut plugin_traces = self.plugins.collect_trace()
-            .into_iter()
-            .map(|json| chrome_trace::decode(&json).unwrap())
-            .collect::<Vec<_>>();
+        let mut plugin_traces_json = self.plugins.collect_trace();
+        let mut plugin_traces = plugin_traces_json.drain(0..)
+            .map(|json| {
+                chrome_trace::decode(json).map_err(|e| {
+                    // TODO(vlovich): transport the name of the plugin along with
+                    // the sample so we can print a better diagnostic about which
+                    // plugin failed to send well-formed JSON.
```

Comment:
one way we could do this is to decode close to the return site; the plugin manager has access to the plugin's name.

Reply:
Can't seem to reply to a top-level review comment so I'm putting it here.  Yes, you observed correctly that trace_dump is itself probably not needed anymore & you can just invoke serde directly.  I'm leaving it for now because trace-ffi depends on it - that dependency could be removed pretty easily but I didn't want the scope to spiral out (I started by removing trace_dump but too many things broke that I felt punting was the correct answer for now).

Reply:
Regarding your comment, yes, we could.  Something to investigate in a future CL I think rather than tackling it all at once.

Reply:
Okay that sounds reasonable. This seems pretty close to ready then, just the CI issue. Happy to take another look whenever it's ready.